pkg/storage/gcp/bucket_cloud.go

```diff
@@ -0,0 +1,79 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/gomods/athens/pkg/config"
+	s "github.com/gomods/athens/pkg/storage"
+	multierror "github.com/hashicorp/go-multierror"
+	"google.golang.org/api/iterator"
+)
+
+// gcpBucket implements the (./pkg/storage/gcp).Bucket interface
+type gcpBucket struct {
```

Comment:
@robjloranger  just an after merge note: in this case it is ok to have iface and impl together

Reply:
I only split them because there will soon be the mock implementation as well.